async-nats/src/jetstream/stream.rs

```diff
@@ -95,6 +97,130 @@ impl Stream {
     pub fn cached_info(&self) -> &Info {
         &self.info
     }
+
+    pub async fn direct_get_next_for_subject_after_sequence(
+        &self,
+        subject: String,
+        sequence: u64,
+    ) -> Result<Message, Error> {
+        let request_subject = format!(
+            "{}.DIRECT.GET.{}",
+            &self.context.prefix, &self.info.config.name
+        );
+        let payload = json!({
+            "seq": sequence,
+            "next_by_subj": subject,
+        });
+
+        let response = self
+            .context
+            .client
+            .request(
+                request_subject,
+                serde_json::to_vec(&payload).map(Bytes::from)?,
+            )
+            .await
+            .map(|message| Message {
+                message,
+                context: self.context.clone(),
+            })?;
+        if let Some(status) = response.status {
+            if let Some(ref description) = response.description {
+                return Err(Box::from(std::io::Error::new(
+                    ErrorKind::Other,
+                    format!("{} {}", status, description),
+                )));
+            }
+        }
+        Ok(response)
+    }
+    pub async fn direct_get_next_for_subject(&self, subject: String) -> Result<Message, Error> {
```

Comment:
Mirrors raw message API

```suggestion
    pub async fn get_next_direct_message_by_subject(&self, subject: String) -> Result<Message, Error> {
```

Reply:
`direct.get` -> `get_direct` is fine. The difference in API is though, that direct supports also `next`, so `message_by_subject` is ambiguous enough, that user would have to read docs to be sure if its next or last.

Reply:
I don't like phrasing `get....direct_message`. It's not a direct message. The name comes from the fact it can be ... fetched from any replica set member. Direct message does not make sense. Its Directly getting a normal Stream Message.